src/main/java/wooteco/subway/station/service/StationService.java

```diff
@@ -0,0 +1,53 @@
+package wooteco.subway.station.service;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import wooteco.subway.station.controller.dto.StationDto;
+import wooteco.subway.station.dao.StationDao;
+import wooteco.subway.station.domain.Station;
+import wooteco.subway.station.domain.StationRepository;
+import wooteco.subway.station.exception.StationException;
+
+@Service
+@Transactional(readOnly = true)
+public class StationService {
+
+    private final StationRepository stationRepository;
+
+    public StationService(final StationDao stationRepository) {
+        this.stationRepository = stationRepository;
+    }
+
+    @Transactional
+    public StationDto save(final Station requestedStation) {
+        final Optional<Station> station = stationRepository.findByName(requestedStation.getName());
+        if (station.isPresent()) {
+            throw new StationException("이미 존재하는 역 이름입니다.");
+        }
+
+        final Station createdStation = stationRepository.save(requestedStation);
+        return StationDto.of(createdStation);
+    }
+
+    public List<StationDto> showAll() {
+        final List<Station> stations = stationRepository.findAll();
+
+        return stations.stream()
+                .map(StationDto::of)
+                .collect(Collectors.toList());
+    }
+
+    @Transactional
+    public void delete(final Long id) {
+        final Optional<Station> station = stationRepository.findById(id);
+        if (station.isPresent()) {
+            stationRepository.delete(id);
+            return;
+        }
+
+        throw new StationException("지우려고 하는 역이 존재하지 않습니다");
```

Comment:
사소한 것이긴 한데.. 예외상황이 발생했을때 오류를 던져주는 부분을 위쪽에서 처리해주면 좋을 것같아요 :) 
정상인 부분을 if안에서 처리 한다면 나중에 로직이 늘어났을때 보기 힘들 것 같아요!

```suggestion
        if (station.isPresent()) {
            throw new StationException("지우려고 하는 역이 존재하지 않습니다");
        }

                stationRepository.delete(id);        
```

Reply:
저도 이 부분을 작성하며 이 위치가 조금 어색해보였어요! 🤔
나중에 로직이 늘어났을 때를 생각하면 확실히 정상적인 부분을 if문 밖으로 빼는 것이 좋겠네요 👍👍